openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,799 @@
-            // Place the category indicator objects into an array
-            var catData = [];
-
-            for (var k in scaledCatIndicator) {
-                catData.push(scaledCatIndicator[k]);
+                indicatorString.push(region + '|'+ theme +'|'+ tempSum);
             }
+        }
+    }
 
-            // TODO remove this concat if need be
-            var concat = catData.concat(previousCatData);
-
-            Category_PCP_Chart(catData, municipality, districName, concat);
-
-            for (var k in scaledCatIndicator) {
-                previousCatData.push(scaledCatIndicator[k]);
-            }
-
-            /////////////////////////////////////////////
-            /////////// Create the svi object ///////////
-            /////////////////////////////////////////////
-
-            for (var k in sviIndicator) {
-                delete sviIndicator[k];
-            }
-
-            for (var k in scaledSviIndicator) {
-                delete scaledSviIndicator[k];
-            }
-
-            for (var k in weightedSviIndicator) {
-                delete weightedSviIndicator[k];
-            }
-
-            function newSVIvalues(SVIaverage, tempMunic) {
-                sviIndicator[tempMunic] = SVIaverage;
-            }
-
-            // Associate the newSVIvalues method with the svi object
-            sviIndicator.SVIvalues = newSVIvalues;
-            var tempSviParentChildKey = [];
-            var tempSviIndicator = {};
-            for (var i = 0; i < municipality.length; i++) {
-                tempSviIndicator[municipality[i]] = [];
-            }
-            tempSviSearchElements = Object.keys(weightedCatIndicator);
-            tempSviSearchElements.pop();
-            var sessionKey = Object.keys(weightedCatIndicator[tempSviSearchElements[0]]);
-            sessionKey.pop();
-            sessionKey.pop();
-
-            for (var i = 0; i < tempSviSearchElements.length; i++) {
-                var tempArray = [];
-
-                // Grab the municipality of the iteration
-                var tempMunic = weightedCatIndicator[i]["municipality"];
-
-                // Remove the municipality from the sessionkey array
-                var index = sessionKey.indexOf("municipality");
-
-                if (index > -1) {
-                    sessionKey.splice(index, 1);
-                }
-
-                // Build the temp object
-                for (var j = 0; j < sessionKey.length; j++) {
-                    tempArray.push(weightedCatIndicator[i][sessionKey[j]]);
-                }
-
-                var SVIsum = 0
-                $.each(tempArray,function() {
-                    SVIsum += this;
-                });
+    for (var p5 = 0; p5 < indicatorString.length; p5++) {
+        // capture an array for each record
+        var indicatorArray;
+        indicatorArray = indicatorString[p5];
+        indicatorArray = indicatorArray.split("|");
```

Comment:
Here, instead of parsing a string, you can get the dictionary from the list and have all the elements easily available.